networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/Eth2Network.java

```diff
@@ -23,4 +24,6 @@
   void unsubscribeFromAttestationSubnetId(final int subnetId);
 
   void setLongTermAttestationSubnetSubscriptions(final Iterable<Integer> subnetIndices);
+
+  void subscribeToAttestations(AttestationManager attestationManager);
```

Comment:
I'd introduce an `AttestationListener` interface and use it here instead of `AttestationManager`.  Actually I think the same interface could be used here and for processed attestations (instead of `ProcessedAttestationConsumer`.  It's the same role for a listener (tell me about attestations) just applied in different contexts (and it's still much more specific than just `Consumer<Attestation>` which I think is useful.

Reply:
I believe you might have gotten this wrong. This method is for the networking layer to subscribe to the attestations processed by `AttestationManager`. The networking layer does not expose its gossip managers so I can't do `AttestationManager.subscribeToProcessedAttestations(attestationGossipManager::onNewAttestation)` at the `BeaconChainController`. I basically need to have access to both `AttestationManager` and `GossipManagers` in the same place and that's how I solved it. Does that make sense?

Reply:
I've actually removed this method, and instead, I now pass in a `Consumer<ProcessedAttestationListener> processedAttestationSubscription` when constructing the `ActiveEth2Network`.  (I might make this a custom interface soon). I set this consumer to `AttestationManager::subscribeToProcessedAttestations`. Then at `ActiveEth2Network` startup, I use this consumer to pass in gossipManagers `onAttestation` methods to the `AttestationManager.subscribeToProcessedAttestations`.